src/outputs/restart_opmd.cpp

```diff
@@ -0,0 +1,180 @@
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2024 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+//! \file restart_opmd.cpp
+//  \brief Restarts a simulation from an OpenPMD output with ADIOS2 backend
+
+#include <iostream>
+#include <memory>
+#include <numeric>
+#include <string>
+#include <vector>
+
+#include "basic_types.hpp"
+#include "interface/params.hpp"
+#include "openPMD/Iteration.hpp"
+#include "openPMD/Series.hpp"
+#include "outputs/parthenon_opmd.hpp"
+#include "outputs/restart.hpp"
+#include "outputs/restart_opmd.hpp"
+#include "utils/error_checking.hpp"
+
+namespace parthenon {
+
+//----------------------------------------------------------------------------------------
+//! \fn void RestartReader::RestartReader(const std::string filename)
+//  \brief Opens the restart file and stores appropriate file handle in fh_
+RestartReaderOPMD::RestartReaderOPMD(const char *filename)
+    : filename_(filename), series(filename, openPMD::Access::READ_ONLY, MPI_COMM_WORLD) {
+  PARTHENON_REQUIRE_THROWS(
+      series.iterations.size() == 1,
+      "Parthenon restarts should only contain one iteration/timestep.");
+  unsigned long idx;
+  for (const auto &i : series.iterations) {
+    idx = i.first;
+  }
+  it = std::make_unique<openPMD::Iteration>(series.iterations[idx]);
+}
+
+int RestartReaderOPMD::GetOutputFormatVersion() const {
+  // TODO(pgrete) move info to shared header and introduce constexpr var
+  if (it->containsAttribute("OutputFormatVersion")) {
+    return it->getAttribute("OutputFormatVersion").get<int>();
+  } else {
+    return -1;
+  }
+}
+
+RestartReaderOPMD::SparseInfo RestartReaderOPMD::GetSparseInfo() const {
+  // TODO(pgrete) needs impl
+  return {};
+}
```

Comment:
so I take it currently OpenPMD does not work with sparse vars?

Reply:
In principle, this should be straightforward to add, since OpenPMD relies on the ADIOS2 sparse support for all AMR blocks anyway (i.e., there are 'sparse' blocks on AMR levels where there is no data at that level, since OpenPMD creates logically-global fields for AMR data).

@pgrete is there something I'm missing that makes this more complicated?

Reply:
Nope, I also expect it to be very straigtforward. As you say, we can just naturally write sparse "data" in bp format but need to add some metainformation on the allocation status.
But that should all be easily done via existing machinery.